lighthouse-core/lib/network-recorder.js

```diff
@@ -341,10 +341,16 @@ class NetworkRecorder extends EventEmitter {
     const records = networkRecorder.getRecords().filter(record => record.isValid);
 
     // create a map of all the records by URL to link up initiator
-    const recordsByURL = new Map();
+    const possiblePrefetchRecords = new Map();
+    const typedRecords = new Map();
```

Comment:
nit: I get why it's named `typedRecords` but it keeps making me think static types. Can I sell you on `nonPrefetchRecords`/`resourceRecords`/`resourceTypedRecords`/`realResourceRecords`?